serving/ingress/pkg/reconciler/ingress/resources/route.go

```diff
@@ -142,6 +142,19 @@ func makeRoute(ci *networkingv1alpha1.Ingress, host string, rule networkingv1alp
 		},
 	}
 
+	// Use passthrough type for OpenShift Ingress -> Kourier Gateawy to encrypt the traffic
+	// when Internal TLS is enabled.
+	// In other words, do not use edge termination which makes plain traffic.
+	if isInternalEncryptionEnabled(rule) {
+		route.Spec.Port = &routev1.RoutePort{
+			TargetPort: intstr.FromString(HTTPSPort),
+		}
+		route.Spec.TLS = &routev1.TLSConfig{
+			Termination:                   routev1.TLSTerminationPassthrough,
```

Comment:
Is this ok if `EnablePassthroughRouteAnnotation` is not there?

Reply:
It creates the same route eventually. So I merged it into the same `if` condition.